### PR TITLE
fix: add theme data source for devtools.

### PR DESCRIPTION
**Pull request by cucbin**
#### Description of Change

Fix: #43367 

The root cause why the theme does not work in DevTools is because there is an error when loading the colors.css file.
[The specific code](https://source.chromium.org/chromium/chromium/src/+/main:third_party/devtools-frontend/src/front_end/ui/legacy/theme_support/ThemeSupport.ts;drc=e6796272dc534345f936479de75d58ca04a8109a;l=207) is:
```
const newColorsCssLink = document.createElement('link');
    newColorsCssLink.setAttribute(
        'href', `devtools://theme/colors.css?sets=ui,chrome&version=${(new Date()).getTime().toString()}`);
    newColorsCssLink.setAttribute('rel', 'stylesheet');
    newColorsCssLink.setAttribute('type', 'text/css');
    newColorsCssLink.onload = () => {
      if (oldColorsCssLink) {
        oldColorsCssLink.remove();
      }
      this.#applyThemeToDocument(document);
    };
```
Because the loading failed, this code `this.#applyThemeToDocument(document);` would not be called, causing the theme to not work.
**In conclusion:** URLDataSource did not process the theme request.

The code reference from: [theme_source.cc](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/webui/theme_source.cc;drc=0f2e551bc3164f7e4f016a93924b83db1b395254;l=98)

If print the logs from DevTools, can see that the colors.css file has been successfully loaded.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Fix theme does not work in DevTools.

### CODE REVIEW

**Comment by cucbin** [comment]
Hi @ckerr, any news on the pull request?

**Review by codebytere — CHANGES_REQUESTED** [review]
@cucbin please properly fill out the template and describe your PR in detail.

- How exactly does this fix the issue?
- Where is this code from? Is it copied from upstream?
- How did you confirm the issue is fixed? Do you have screenshots?
- Can a test be added for this? If not, why not?

File: shell/browser/ui/devtools_ui.cc
Comment (codebytere) [inline shell/browser/ui/devtools_ui.cc:143]: I'd prefer if we followed the file conventions in Chromium here and made this a separate file like [devtools_ui_data_source.cc](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/webui/devtools/devtools_ui_data_source.cc)

Comment (cucbin) [reply shell/browser/ui/devtools_ui.cc:143]: Do you mean both `BundledDataSource `and `ThemeDataSource ` need to be separated into separate files?

Comment (samuelmaddock) [reply shell/browser/ui/devtools_ui.cc]: I had some initial confusion reading through this so I think separate files like `devtools_ui_bundled_source` and `devtools_ui_theme_source` could be helpful. Especially given that [theme_source.h](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/webui/theme_source.h;drc=36147a880112f3dfd7a288309c83a483b520dd7c) is its own file upstream.

Comment (codebytere) [reply shell/browser/ui/devtools_ui.cc]: @cucbin yes - I agree with @samuelmaddock here.

Comment (cucbin) [reply shell/browser/ui/devtools_ui.cc]: I have considered this before, but considering that there are duplicate code of anonymous methods, so final put them in one file.
But I also agree this, had changed it.

File: shell/browser/ui/devtools_ui_theme_data_source.h
Comment (samuelmaddock) [inline shell/browser/ui/devtools_ui_theme_data_source.h]:
```suggestion
#ifndef ELECTRON_SHELL_BROWSER_UI_DEVTOOLS_UI_THEME_DATA_SOURCE_H_
#define ELECTRON_SHELL_BROWSER_UI_DEVTOOLS_UI_THEME_DATA_SOURCE_H_
```

File: shell/browser/ui/devtools_ui_theme_data_source.h
Comment (samuelmaddock) [inline shell/browser/ui/devtools_ui_theme_data_source.h:4]: Typically this is not used in Chromium code.
```suggestion
```

File: shell/browser/ui/devtools_ui_bundle_data_source.h
Comment (samuelmaddock) [inline shell/browser/ui/devtools_ui_bundle_data_source.h:4]:
```suggestion
```

**Review by samuelmaddock — CHANGES_REQUESTED** [review]
a few minor requests, lgtm after making these 👍

**Review by samuelmaddock — APPROVED** [review]

**Comment by bpasero** [comment]
Yes please 🙏

**Review by codebytere — APPROVED** [review]

**Comment by cucbin** [comment]
Hi @codebytere @samuelmaddock  Could you guys help me to merge it? It shows that I am not authorized to merge this PR. 
There is a patch conflict to 32-x-y, should I also continue to handle it in #44098?

**Comment by codebytere** [comment]
@cucbin please rebase this PR! Once that's done  we can merge this. I'd wait until it's merged and then update the backport pull request to 32.

**Comment by cucbin** [comment]
> @cucbin please rebase this PR! Once that's done we can merge this. I'd wait until it's merged and then update the backport pull request to 32.

Hi, rebase has been processed.

**Pull request merged**
